ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobMetadata.java

```diff
@@ -399,8 +420,72 @@ public boolean equals(Object o) {
       return false;
     }
     CloudBlobMetadata om = (CloudBlobMetadata) o;
-    return (Objects.equals(id, om.id) && Objects.equals(partitionId, om.partitionId) && creationTime == om.creationTime
+    return (Objects.equals(id, om.id) && Objects.equals(partitionId, om.partitionId) && accountId == om.accountId
+        && containerId == om.containerId && size == om.size && creationTime == om.creationTime
         && uploadTime == om.uploadTime && expirationTime == om.expirationTime && deletionTime == om.deletionTime
-        && size == om.size && accountId == om.accountId && containerId == om.containerId);
+        && namingScheme == om.namingScheme && encryptionOrigin == om.encryptionOrigin && Objects.equals(vcrKmsContext,
+        om.vcrKmsContext) && Objects.equals(cryptoAgentFactory, om.cryptoAgentFactory)
+        && encryptedSize == om.encryptedSize);
+  }
+
+  /**
+   * @return a {@link HashMap} of metadata key-value pairs.
+   */
+  public HashMap<String, String> toMap() {
+    ObjectMapper objectMapper = new ObjectMapper();
+    HashMap<String, String> map = new HashMap<>();
+    ObjectNode node = objectMapper.convertValue(this, ObjectNode.class);
+    node.fieldNames().forEachRemaining(fieldName -> {
+      map.put(fieldName, node.get(fieldName).asText());
+    });
+    return map;
+  }
+
+  /** Custom serializer for CloudBlobMetadata class that omits fields with default values. */
+  static class MetadataSerializer extends StdSerializer<CloudBlobMetadata> {
+
+    public MetadataSerializer() {
+      this(null);
+    }
+
+    public MetadataSerializer(Class<CloudBlobMetadata> t) {
```

Comment:
could you explain the purpose of these constructors? Would the value passed in always be `CloudBlobMetadata.class`?

Reply:
I removed this constructor.  It was in an example I looked at but doesn't seem to be needed here.  I guess it's used for base classes that have multiple derived classes with different serializations?